### PR TITLE
config from azure key vault

### PR DESCRIPTION
**Pull request by FelixKuhnAnsys**
includes new standard InitConfig function for allie go modules

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -1,17 +1,115 @@
...
 	"time"
 
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azsecrets"
 	"gopkg.in/yaml.v2"
 )
 
+////////////////////////////////////////////
+// Stanard Config init for Allie Go Modules
```
Comment (MaxJPRey) [inline pkg/config/config.go:22]:
```suggestion
// Standard Config init for Allie Go Modules
```

File: pkg/config/config.go [hunk 2]
```diff
@@ -168,10 +227,10 @@ func isZeroValue(v reflect.Value) bool {
 //
 // Parameters:
 //   - fileName: The name of the configuration file.
```
Comment (AlejandroFernandezLuces) [inline pkg/config/config.go:229]: Return documentation is missing here.

File: pkg/config/config.go [hunk 3]
```diff
@@ -259,10 +312,152 @@ func createFlags(val reflect.Value, prefix string) {
...
 ///////////////////////
 // Helper Functions
 ///////////////////////
 
+// ValidateConfig checks for mandatory entries in the configuration and validates chosen models.
+//
+// Parameters:
+//   - config: The configuration object to validate.
+//   - requiredProperties: The list of required properties.
+func ValidateConfig(config Config, requiredProperties []string) (err error) {
```
Comment (AlejandroFernandezLuces) [inline pkg/config/config.go:445]: Same here

File: pkg/config/config.go [hunk 4]
```diff
@@ -22,38 +120,36 @@ import (
...
+	}
+
+	return nil
 }
 
 // readYaml reads the yaml specified in `fileName` parameter and saves it to `config_struct`
 //
 // Parameters:
 //   - fileName: The name of the configuration file.
 //   - configStruct: Struct with the parameters of the YAML to read.
```
Comment (AlejandroFernandezLuces) [inline pkg/config/config.go:146]: Same

**Review by RobPasMue — APPROVED** [review]

File: pkg/config/config.go [hunk 5]
```diff
@@ -22,38 +120,36 @@ import (
 //   - fileName: The name of the configuration file.
 //   - requiredProperties: The list of required properties.
 //   - optionalDefaultValues: The map of optional properties and their default values.
```
Comment (AlejandroFernandezLuces) [inline pkg/config/config.go:122]: Missing return docs here too

**Review by AlejandroFernandezLuces — APPROVED** [review]
LGTM, some minor comments regardind docstrings

File: pkg/config/config.go (hunk 2 above)
Comment (FelixKuhnAnsys) [reply pkg/config/config.go:239]: thanks!

**Pull request merged**